code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -54,7 +54,6 @@
 		var/obj/item/stack/S = O
 
 		if(!S)
-			modules -= null

```

Comment:
This was a stupidly cheesy way of clearing out nulls from a list **THAT SHOULD HAVE BEEN CLEARED OUT PROPERLY IN THE FIRST PLACE**